simpleclient_common/src/main/java/io/prometheus/client/exporter/common/TextFormat.java

```diff
@@ -19,11 +20,11 @@ public static void write004(Writer writer, Enumeration<Collector.MetricFamilySam
       writer.write("# TYPE " + metricFamilySamples.name + " " + typeString(metricFamilySamples.type) + "\n");
       for (Collector.MetricFamilySamples.Sample sample: metricFamilySamples.samples) {
         writer.write(sample.name);
-        if (sample.labelNames.size() > 0) {
+        if (sample.labels != null && !sample.labels.isEmpty()) {
           writer.write("{");
-          for (int i = 0; i < sample.labelNames.size(); ++i) {
+          for (Map.Entry<String, String> label : sample.labels.entrySet()) {
```

Comment:
The output can be sorted, but this extra operation won't improve efficiency. :-)

I just had a look at the [Python client](https://github.com/prometheus/client_python/blob/master/prometheus_client/exposition.py#L64) and it outputs sorted labels, so we could do the same for better consistency, if you like the idea.